commands/build.go

```diff
@@ -127,16 +137,9 @@ func runBuild(cmd *cobra.Command, args []string) error {
 		if parsedServices != nil {
 			services = *parsedServices
 		}
-	}
 
-	if pullErr := PullTemplates(DefaultTemplateRepository); pullErr != nil {
-		return fmt.Errorf("could not pull templates for OpenFaaS: %v", pullErr)
-	}
-
-	if len(services.Functions) > 0 {
 		build(&services, parallel, shrinkwrap)
-
-	} else {
+	} else if functionSource == "inline" {
```

Comment:
If we're going to use hard-coded strings `"inline"` and `"YAML"` then these should be abstracted away in constants. I would call it:

```
const argumentSource = "arguments"
const  YAMLSource = "yamlFile"
```

For instance.

Reply:
Agree with Alex's comment.  The strings are also used in too many places not to be constants.